indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,659 @@
+        if (nonbookablePeriods || bookableHours) {
+            return indicoAxios.post(updateRoomAvailabilityURL({room_id: roomId}), snakifyKeys(_.pick(changedValues, ['bookableHours', 'nonbookablePeriods'])));
+        } else {
+            return Promise.resolve();
+        }
+    };
+
+    renderImage = (position) => {
+        return <SpriteImage key="image" pos={position} />;
+    };
+
+    renderAttributes = (content) => {
+        const {attributes, roomAttributes} = this.state;
+        if (!attributes || !roomAttributes) {
+            return null;
+        }
+        const titles = _.fromPairs(attributes.map(x => [x.name, x.title]));
+        return (
+            <FieldArray key="attributes" name="attributes" isEqual={_.isEqual}>
+                {({fields}) => {
+                    if (!fields.value) {
+                        return null;
+                    }
+                    const options = this.generateAttributeOptions(attributes, fields.value);
+                    return (
+                        <div>
+                            <Dropdown className="icon room-edit-modal-add-btn"
+                                      button
+                                      text={content.placeholder}
+                                      floating
+                                      labeled
+                                      icon="add"
+                                      options={options}
+                                      search
+                                      disabled={options.length === 0}
+                                      selectOnBlur={false}
+                                      onChange={(__, {value: name}) => {
+                                          fields.push({value: null, name});
+                                      }} />
+                            {fields.map((attribute, index) => (
+                                <div key={attribute}>
+                                    <Field label={titles[fields.value[index].name]}
+                                           name={`${attribute}.value`}
+                                           component={ReduxFormField}
+                                           as={Input}
+                                           isEqual={_.isEqual}
+                                           required
+                                           icon={{name: 'trash', color: 'red', link: true, onClick: () => fields.remove(index)}} />
+                                </div>
+                            ))}
+                        </div>
+                    );
+                }}
+            </FieldArray>
+        );
+    };
+
+    renderColumn = (column, key) => (
+        <Grid.Column key={key}>{column.map(this.renderContent)}</Grid.Column>
+    );
+
+    renderContent = (content, key) => {
+        const {room, roomEquipment, submitSucceeded} = this.state;
+        const {equipmentTypes, favoriteUsers} = this.props;
```

Comment:
The same I guess as above? (`![] === false`)

Reply:
Ok, I think I might not be right here since as far as I can tell `equipmentTypes` is an object

Reply:
Oh, but still `!{}` suffers from the same problem